graylog2-server/src/main/java/org/graylog/plugins/views/search/validation/FieldTypeValidationImpl.java

```diff
@@ -89,6 +90,24 @@ private static boolean isDate(final String dateCandidate) {
         return false;
     }
 
+
+    private static boolean isValidIp(final String ip) {
+        try {
+            return InetAddressValidator.getInstance().isValid(ip) || isValidCidrIp(ip);
+        } catch (Exception ex) {
+            return false;
+        }
+    }
+
+    private static boolean isValidCidrIp(String cidrIp) {
+        try {
+            new SubnetUtils(cidrIp);
+            return true;
+        } catch (IllegalArgumentException e) {
+            return false;
+        }
+    }
```

Comment:
I'd consider moving it to separate class, as the responsibility of this class has grown to multiple different types of validation. But it is not a must...

Reply:
I think the class is still reasonably short and focused on one thing, I'd keep the method there now. If it grows significantly in the future, or we'll have the need to reuse the IP validation elsewhere, then we can always extract it.